src/data/history.ts

```diff
@@ -200,6 +201,23 @@ const processLineChartEntities = (
   };
 };
 
+const isNumerical = (states: HassEntity[]): boolean => {
+  if (states.every((state) => UNAVAILABLE_STATES.includes(state.state))) {
+    return false;
+  }
+
+  if (
+    states.some(
```

Comment:
We are now going over all states twice, can we do this in one go?

Reply:
I took a walk to think about this, and I don't think so.

I tried turning the logic inside out to make both checks be a `some` or an `every`, and I don't think it can be done.

The remaining alternative is to loop through all states manually, with one early return and one sentry variable, like
```js
let allUnavailable = true;
for (const state of states) {
  if (
    isNaN(parseFloat(state.state)) &&
    !UNAVAILABLE_STATES.includes(state.state)
  ) {
    return false;
  }
  if (!UNAVAILABLE_STATES.includes(state.state)) {
    allUnavailable = false;
  }
}
if (allUnavailable) {
  return false;
}
```

The number of comparison operations remains the same, and I think the old version is more clear.

But if you like this better, I'll be happy to change.

Reply:
I do like this better, as it gives us an early escape